core/src/avm1/activation.rs

```diff
@@ -988,16 +988,21 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     fn action_enumerate(&mut self) -> Result<FrameControl<'gc>, Error<'gc>> {
         let name_value = self.context.avm1.pop();
         let name = name_value.coerce_to_string(self)?;
-        self.context.avm1.push(Value::Null); // Sentinel that indicates end of enumeration
```

Comment:
Actually, it seems like `Value::Undefined` should be pushed here. Consider:
```
Push "hello" "a" 1 "b" 2 2
InitObject
Enumerate2
Trace
Trace
Trace
Trace
Trace
Trace
```
Trace in Flash Player:
```
a
b
undefined
hello
undefined
undefined
```
Notice that `Value::Null` is never pushed, but instead a `Value::Undefined` senitel.

`Value::Undefined` should be pushed for non-Objects too:
```
Push "hello" 123
Enumerate2
Trace
Trace
Trace
Trace
Trace
Trace
```
Trace in Flash Player:
```
undefined
hello
undefined
undefined
undefined
undefined
```

This demonstrates `Enumerate2` but applies to `Enumerate` as well, so both `action_enumerate` and `action_enumerate_2` should be updated.

Reply:
I didn't consider that this might also apply to enumerate2 as well, looks like in all cases the sentinel is `Undefined` rather than `null`. I've update the tests to check enumerate2 as well as `Objects`